gamma/utils.py

```diff
@@ -274,12 +305,20 @@ def associate(
                 continue
 
         if lock is not None:
-            lock.acquire()
-
-        if not isinstance(event_idx, int):
-            event_idx_value = event_idx.value
+            with lock:
+                if not isinstance(event_idx, int):
+                    event_idx.value += 1
+                    event_idx_value = event_idx.value
+                else:
+                    event_idx += 1
+                    event_idx_value = event_idx
         else:
-            event_idx_value = event_idx
+            if not isinstance(event_idx, int):
+                event_idx.value += 1
```

Comment:
Is it better to put with lock here? so that you do not need to repeat a few lines of code.

Reply:
It can ensure the lock is always released using `with lock`, even though there's an error raised.